dask/base.py

```diff
@@ -1580,4 +1582,49 @@ def clone_key(key, seed):
     raise TypeError(f"Expected str or tuple[str, Hashable, ...]; got {key}")
 
 
+def print_on_compute(msg: str, val: T) -> T:
+    """When the value is computed, print it using distributed.print.
+
+    Note: this may affect performance, since it will change the graph being
+    computed.
+
+    Parameters
+    ----------
+    msg : str
+          Format string for printing out value. Default "%s".
+          It's also possible to use placeholders compatible with str.format(),
+          such as "{}" or "{0}", or the named placeholder "{result}"
+    val : T
+          Lazy value, including a Dask collection, Scalar, etc.
+
+    Returns
+    -------
+    result : T
+          Original value wrapped with an additional Delayed layer
+    """
+    msg = msg or "%s"
+
+    import builtins
+
+    from dask import delayed
+
+    dask_key_name = f"{list(get_collection_names(val))[0]}-print"
```

Comment:
this is a `list` of a `set`. The order of sets is not deterministic so this will generate a different result every time you restart your interpreter, e.g.

```
In [23]: list(set(list("abcdefg")))
Out[23]: ['a', 'd', 'b', 'c', 'f', 'e', 'g']

In [24]:
Do you really want to exit ([y]/n)? ^D
                                                                                                               1d8h ◒
▶ ipython
Python 3.10.11 | packaged by conda-forge | (main, May 10 2023, 19:01:19) [Clang 14.0.6 ]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.14.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: list(set(list("abcdefg")))
Out[1]: ['b', 'd', 'f', 'g', 'c', 'a', 'e']

```

Reply:
That's actually a question I had. I tried this method on several different Dask instances (Dataframe, Array, Delayed), and I always get a single value back. Are there ever cases when this is not a single value for a single Dask object?

Reply:
Unlikely but not impossible. The API does not restrict what custom collections can use as keynames so we should not encode any kind of assumptions in here. I also don't know if there are not any kind of special reductions that return something more complex

Reply:
Ok. I changed it to retrieve the first collection name.